javascript/modules/client/src/client-erc20.ts

```diff
@@ -42,51 +47,131 @@ export class ClientErc20 extends ClientCore implements IClientErc20 {
 
   /** Contains all the generic high level methods to interact with a DAO */
   methods = {
-    createProposal: (params: ICreateProposalParams) =>
+    /**
+     * Creates a new ERC20 voting proposal
+     *
+     * @param {ICreateProposalParams} params
+     * @return {*}  {AsyncGenerator<ProposalCreationStepValue>}
+     * @memberof ClientErc20
+     */
+    createProposal: (params: ICreateProposalParams): AsyncGenerator<ProposalCreationStepValue> =>
       this._createProposal(params),
-    voteProposal: (proposalId: string, vote: VoteOptions) =>
+    /**
+     * Cast a vote o a proposal given the proposalId and the vote
+     *
+     * @param {string} proposalId
+     * @param {VoteOptions} vote
+     * @return {*}  {AsyncGenerator<VoteProposalStepValue>}
+     * @memberof ClientErc20
+     */
+    voteProposal: (proposalId: string, vote: VoteOptions): AsyncGenerator<VoteProposalStepValue> =>
       this._voteProposal(proposalId, vote),
-    executeProposal: (proposalId: string) => this._executeProposal(proposalId),
-    // setDaoConfig: (address: string, config: DaoConfig) =>
-    //   this._setDaoConfig(address, config),
-    // setVotingConfig: (address: string, config: VotingConfig) =>
-    //   this._setVotingConfig(address, config),
-    /** Retrieves the list of Ethereum addresses that can cast weighted votes on the given DAO's proposals */
-    getMembers: (daoAddressOrEns: string) => this._getMembers(daoAddressOrEns),
-    /** Retrieves the list of proposals of the given DAO */
-    getProposals: (daoAddressOrEns: string) =>
-      this._getProposals(daoAddressOrEns),
+    /**
+     * Executes a proposal given a specific proposalId
+     *
+     * @param {string} proposalId
+     * @return {*}  {AsyncGenerator<ExecuteProposalStepValue>}
+     * @memberof ClientErc20
+     */
+    executeProposal: (proposalId: string): AsyncGenerator<ExecuteProposalStepValue> =>
+      this._executeProposal(proposalId),
+    /**
+     * Sets the voting configuration in a multisig proposal given a daoAddress and a configuration
+     *
+     * @param {string} daoAddressOrEns
+     * @param {VotingConfig} config
+     * @return {*}  {AsyncGenerator<SetVotingConfigStepValue>}
+     * @memberof ClientErc20
+     */
+    setVotingConfig: (daoAddressOrEns: string, config: VotingConfig): AsyncGenerator<SetVotingConfigStepValue> =>
+      this._setVotingConfig(daoAddressOrEns, config),
+
+    /**
+     * Returns the list of members given a dao address or ens
+     *
+     * @param {string} daoAddressOrEns
```

Comment:
I'm not sure we need this parameter. The token contract address would be needed, but it's likely that the contract already knows this, so probably not needed.